Chroma + Mapping extensions support

VSjnk opened the pull request:
My pull request adds Chroma notes, walls, lights support and a somewhat unfinished mapping extensions support.
The only real issue is that mapping extenisons walls don't work 100% correctly yet.

VSjnk committed:
Updated to godot 4.4

VSjnk committed:
added partial mapping extensions support

Walls are currently funky

VSjnk committed:
Fixed note stuff

Fixed a error I had moving over old code I made for a previous version of open saber

VSjnk committed:
added more chroma support

lights are now more customizable

VSjnk committed:
Added/fixed more mapping extension features

Fixed a miss spelling of a constant due to me copying over code and updating it.

Walls have been improves

VSjnk committed:
Update Wall.gd

More attempts to make walls work, heights cannot be placed consitantly

VSjnk committed:
Update Wall.gd

VSjnk committed:
Attempts to recreate the original mods handeling

Im just translating the og mapping extensions AdditionalWallAdjustmentsPatches.cs script

VSjnk committed:
Update README.md

VSjnk committed:
Update README.md

VSjnk committed:
Update README.md

VSjnk committed:
Wall improvements!

Walls are still geeked up, but with my skibidi coding skills I have figured out the placement!!!

VSjnk committed:
Update Wall.gd

this walls arn't very sigma

VSjnk committed:
fixed some wall width stuff

VSjnk committed:
Update README.md

VSjnk commented:
This is proff that I have a optimized method of removing blocks, but I need to properly port it over
https://www.youtube.com/watch?v=pLvrPLhQb5Y

enmityVR commented:
any chance of noodle being an option at some point? it would be pretty neat.

ParlDev commented:
This is pretty awesome, but I would like to make sure that we check that the [README](https://github.com/leandrodreamer/BeepSaber/blob/23e588a0515ff50ab01158c17766534e9a799772/README.md) doesn't change this much, especially at the top. We should change it slightly, but not to this degree.

VSjnk commented:
> any chance of noodle being an option at some point? it would be pretty neat.

I want to add noodle extensions, and I have some code that does work, but because open saber is made in godot, and noodle extensions is in unity, I have to make some changes to the order of the cordinates.

but if I have enough time, noodle extensions will be added.

VSjnk commented:
> This is pretty awesome, but I would like to make sure that we check that the [README](https://github.com/leandrodreamer/BeepSaber/blob/23e588a0515ff50ab01158c17766534e9a799772/README.md) doesn't change this much, especially at the top. We should change it slightly, but not to this degree.

don't blame you, I changed it like that because I wanted a way to track my progress and make it easily readable to anyone, It's more really for anyone who stumbles upon it, but I can make a less 'talk about the additions' version

ParlDev commented:
> don't blame you, I changed it like that because I wanted a way to track my progress and make it easily readable to anyone, It's more really for anyone who stumbles upon it, but I can make a less 'talk about the additions' version

That makes sense. I think that you have done an awesome job on all of this. But yeah, I would instead just say that it supports Chroma and Mapping Extensions somewhere in the readme, and you should be good. Hopefully @leandrodreamer will accept it when finished.

VSjnk committed:
Update README.md

Made the "Mods" Section out of the way, hopefully so it can be more compliant with being merged with the main project

ParlDev commented:
Lol the note for vivify support being just "No." made me laugh.
That would basically be impossible to implement unfortunately, although open source has surprised me before.

In all seriousness that looks pretty good. We just need @leandrodreamer to approve it now.
